speech/cloud-client/src/main/java/com/example/speech/Recognize.java

```diff
@@ -830,4 +839,136 @@ public static void transcribeModelSelectionGcs(String gcsUri) throws Exception {
+              SpeechRecognitionAlternative alternative = result.getAlternativesList().get(0);
+              System.out.printf("Transcript : %s\n", alternative.getTranscript());
+            }
+
+            public void onComplete() {}
+
+            public void onError(Throwable t) {
+              System.out.println(t);
+            }
+          };
+
+      ClientStream<StreamingRecognizeRequest> clientStream =
+          client.streamingRecognizeCallable().splitCall(responseObserver);
+
+      RecognitionConfig recognitionConfig =
+          RecognitionConfig.newBuilder()
+              .setEncoding(RecognitionConfig.AudioEncoding.LINEAR16)
+              .setLanguageCode("en-US")
+              .setSampleRateHertz(16000)
+              .build();
+      StreamingRecognitionConfig streamingRecognitionConfig =
+          StreamingRecognitionConfig.newBuilder().setConfig(recognitionConfig).build();
+
+      StreamingRecognizeRequest request =
+          StreamingRecognizeRequest.newBuilder()
+              .setStreamingConfig(streamingRecognitionConfig)
+              .build(); // The first request in a streaming call has to be a config
+
+      clientStream.send(request);
+
+      try {
+        // SampleRate:16000Hz, SampleSizeInBits: 16, Number of channels: 1, Signed: true,
+        // bigEndian: false
+        AudioFormat audioFormat = new AudioFormat(16000, 16, 1, true, false);
+        DataLine.Info targetInfo =
+            new Info(
+                TargetDataLine.class,
+                audioFormat); // Set the system information to read from the microphone audio stream
+
+        if (!AudioSystem.isLineSupported(targetInfo)) {
+          System.out.println("Microphone not supported");
+          System.exit(0);
+        }
+        // Target data line captures the audio stream the microphone produces.
+        targetDataLine = (TargetDataLine) AudioSystem.getLine(targetInfo);
+        targetDataLine.open(audioFormat);
+        micThread.start();
+
+        long startTime = System.currentTimeMillis();
+
+        while (true) {
+
+          long estimatedTime = System.currentTimeMillis() - startTime;
+
+          if (estimatedTime >= 55000) {
+
+            clientStream.closeSend();
+            clientStream = client.streamingRecognizeCallable().splitCall(responseObserver);
+
+            request =
+                StreamingRecognizeRequest.newBuilder()
+                    .setStreamingConfig(streamingRecognitionConfig)
+                    .build();
+
+            startTime = System.currentTimeMillis();
+
+          } else {
+            request =
+                StreamingRecognizeRequest.newBuilder()
+                    .setAudioContent(ByteString.copyFrom(sharedQueue.take()))
+                    .build();
+          }
+
+          clientStream.send(request);
+        }
+      } catch (Exception e) {
+        System.out.println(e);
+      }
+      responseObserver.onComplete();
+
+      clientStream.closeSend();
```

Comment:
Lines 968-970
If we do a "Ctrl-C" to exit, these lines won't be called, right?
Is there a way to still exit and call them?

Reply:
Removed the `responseObserver.onComplete()` since I haven't really defined any particular action that needs to be performed. Checked the `clientStream` functionality and found out it gets closed by default when we hit `Ctrl-C` . Tested it to make sure. Yay! Please let me know if there maybe anything else that is necessary.